templates/articles/blocks/promoted_pages.html

```diff
@@ -29,7 +29,7 @@ <h3 class="promoted-pages__title">
                         {{ promoted_item.title }}
                     </a>
                 </h3>
-                <p class="promoted-pages__text promoted-pages__text--gap">By {{ promoted_item.author }} - {{ promoted_item.publication_date }}</p>
+                <p class="promoted-pages__text promoted-pages__text--gap">{% if promoted_item.author %}By {{ promoted_item.author }} {% if promoted_item.publication_date %}–{% endif%}{% endif %} {% if promoted_item.publication_date %}{{ promoted_item.publication_date }}{% endif %}</p>
```

Comment:
What's the thinking behind the multiple `{% if %}` conditions here, @jamesbiggs. Should we not just keep it simple and do it with one?